satpy/demo/viirs_sdr.py

```diff
@@ -0,0 +1,425 @@
+    "I02": SVI02_FILES,
+    "I03": SVI03_FILES,
+    "I04": SVI04_FILES,
+    "I05": SVI05_FILES,
+    "M01": SVM01_FILES,
+    "M02": SVM02_FILES,
+    "M03": SVM03_FILES,
+    "M04": SVM04_FILES,
+    "M05": SVM05_FILES,
+    "M06": SVM06_FILES,
+    "M07": SVM07_FILES,
+    "M08": SVM08_FILES,
+    "M09": SVM09_FILES,
+    "M10": SVM10_FILES,
+    "M11": SVM11_FILES,
+    "M12": SVM12_FILES,
+    "M13": SVM13_FILES,
+    "M14": SVM14_FILES,
+    "M15": SVM15_FILES,
+    "M16": SVM16_FILES,
+}
+
+
+def get_viirs_sdr_20170128_1229(
+        base_dir=None,
+        channels=("I01", "I02", "I03", "I04", "I05",
+                  "M01", "M02", "M03", "M04", "M05", "M06", "M07", "M08", "M09", "M10",
+                  "M11", "M12", "M13", "M14", "M15", "M16",
+                  "DNB"),
+        num_granules=10):
+    r"""Get VIIRS SDR files for 2017-01-28 12:29 to 12:43.
+
+    These files are downloaded from Zenodo. You can see the full file
+    listing here: https://zenodo.org/record/263296
+
+    Specific channels can be specified with the ``channels`` keyword argument.
+    By default, all channels (all I bands, M bands, and DNB bands) will be
+    downloaded. Channels are referred to by their band type and channel number
+    (ex. "I01" or "M16" or "DNB"). Terrain-corrected geolocation files are
+    always downloaded when the corresponding band data is specified.
+
+    The ``num_granules`` argument will control how many granules ("time steps")
+    are downloaded. There are 10 available.
+
+    This full dataset is ~10.1GB.
+
+    Notes:
+        File list was retrieved using the zenodo API.
+
+        .. code-block:: python
+
+            import requests
+            viirs_listing = requests.get("https://zenodo.org/api/records/263296")
+            viirs_dict = json.loads(viirs_listing.content)
+            print("\n".join(sorted(x['links']['self'] for x in viirs_dict['files'])))
+
+    """
+    base_dir = base_dir or config.get("demo_data_dir", ".")
+
+    # assume directory in zip is the same as zip filename without the extension
+    subdir = os.path.join(base_dir, "viirs_sdr", "20170128_1229")
+    os.makedirs(subdir, exist_ok=True)
+    urls = (ZENODO_BASE_URL + fn for fn in _get_filenames_to_download(channels, num_granules))
+
+    files = []
+    for url in urls:
+        target = os.path.join(subdir, os.path.basename(url))
+        files.append(target)
+        if os.path.isfile(target):
+            logger.info(f"File {target} already exists, skipping...")
+            continue
+        logger.info(f"Downloading file to {target}...")
+        _download_url(url, target)
+
+    return files
+
+
+def _get_filenames_to_download(channels, num_granules):
+    if any("DNB" in chan for chan in channels):
+        yield from GDNBO_URLS[:num_granules]
+    if any("I" in chan for chan in channels):
+        yield from GITCO_URLS[:num_granules]
+    if any("M" in chan for chan in channels):
+        yield from GMTCO_URLS[:num_granules]
```

Comment:
From experience, the first and last granules aren't always the best. I haven't checked recently with that particular dataset, so I don't remember how it is. However, for flexibility, I was thinking that instead giving the number of granules, we could pass the list of granule numbers to use?

Reply:
I think for VIIRS it only makes sense to request contiguous granules. Maybe someone could pass a slice object as an option?

Reply:
I wouldn't worry too much about non-contiguous granules. I happens that some data get corrupted during reception, so that in the end we just have a missing granule in the middle. It's actually a good test (nothing we should implement here) to see how the resampling works in that case...